### PR TITLE
187 tcp ipc from a new module in stepbro module name and outside app name yet to be decided

### PR DESCRIPTION
**Pull request by MolaeMaceria**
Move SideKickPipelines to IPCPipe.
Fix issues that arise from this.
Clean up a bit in usings.

Add ExecutionHelper, make it possible to create, increment, set, and get variables through it, as well as save these to a file and load it from a file.

Create an event handler for the pipe, so a timer is not necessary.

### CODE REVIEW

File: source/StepBro.Core/IPCPipe/IPCPipe.cs [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
-using System.IO;
 using System.IO.Pipes;
+using System.IO;
 using System.Security.Principal;
 using System.Text;
 using System.Text.Json;
 using System.Threading;
 
-namespace StepBro
+namespace StepBro.Core.IPCPipe
```
Comment (JanJorgensen) [inline source/StepBro.Core/IPCPipe/IPCPipe.cs:10]: Class and namespace with same name is not a good idea. Maybe namespace IPC and class Pipe.

File: source/Modules/StepBro.ExecutionHelper.Access/Access.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                }
+            }
+        }
+
+        public bool CreateExecutionHelper(bool closeWhenExecutionHelperCloses = false)
+        {
+            m_closeWhenExecutionHelperCloses = closeWhenExecutionHelperCloses;
+            bool result = true;
+
+            var hThis = GetConsoleWindow();
```
Comment (JanJorgensen) [inline source/Modules/StepBro.ExecutionHelper.Access/Access.cs:37]: This is not necessary.

File: source/Modules/StepBro.ExecutionHelper.Access/Access.cs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+            string path = Assembly.GetExecutingAssembly().Location;
+            var folder = Path.GetDirectoryName(path);
+
+            string pipename = hThis.ToString("X");
+            m_executionHelperPipe = Pipe.StartServer("StepBroExecutionHelper", pipename);
```
Comment (JanJorgensen) [inline source/Modules/StepBro.ExecutionHelper.Access/Access.cs:43]: This should be the client, I think.

File: source/Modules/StepBro.ExecutionHelper.Access/Access.cs [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+using System.Threading;
+
+namespace StepBro.ExecutionHelper
+{
+    public class Access
+    {
+        private Pipe m_executionHelperPipe = null;
+        bool m_executionHelperStarted = false;
+        bool m_closeWhenExecutionHelperCloses = false;
+        EventHandler<Tuple<string, string>> receivedDataEventHandler = null;
```
Comment (JanJorgensen) [inline source/Modules/StepBro.ExecutionHelper.Access/Access.cs:16]: Is it necessary to save this?

File: source/Modules/StepBro.ExecutionHelper.Access/Access.cs [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+            string path = Assembly.GetExecutingAssembly().Location;
+            var folder = Path.GetDirectoryName(path);
+
+            string pipename = hThis.ToString("X");
+            m_executionHelperPipe = Pipe.StartServer("StepBroExecutionHelper", pipename);
+            var executionHelper = new System.Diagnostics.Process();
+            executionHelper.StartInfo.FileName = Path.Combine(folder, "../StepBro.ExecutionHelper.exe"); //../ because ExecutionHelper is in the main bin folder and this is the Modules folder
+            executionHelper.StartInfo.Arguments = pipename;
+            m_executionHelperStarted = executionHelper.Start();
```
Comment (JanJorgensen) [inline source/Modules/StepBro.ExecutionHelper.Access/Access.cs:47]: Maybe the helper is started already, so try to connect first (as slave) and then start Helper app if it is not possible to connect.

File: source/Modules/StepBro.ExecutionHelper.Access/Access.cs [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+            }
+            else
+            {
+                result = false;
+            }
+
+            return result;
+        }
+
+        public bool CreateVariable(string variableName, object initialValue)
```
Comment (JanJorgensen) [inline source/Modules/StepBro.ExecutionHelper.Access/Access.cs:74]: Maybe this should be named "CreateOrSetVariable".

File: source/Modules/StepBro.ExecutionHelper.Access/Access.cs [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+        }
+
+        public object GetVariable(string variableName)
+        {
+            m_executionHelperPipe.Send(new StepBro.ExecutionHelper.Messages.GetVariable(variableName));
+
+            object variable = -1;
+
+            int timeoutMs = 2500;
+            var input = m_executionHelperPipe.TryGetReceived();
```
Comment (JanJorgensen) [inline source/Modules/StepBro.ExecutionHelper.Access/Access.cs:105]: Just set to null (DRY).

File: source/StepBro.ExecutionHelper/ExecutionHelperMessages.cs [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+{
+    internal class ExecutionHelperMessages
+    {
+    }
+
+    public enum ShortCommand
+    {
+        None,
+        Close,
+        Acknowledge
```
Comment (JanJorgensen) [inline source/StepBro.ExecutionHelper/ExecutionHelperMessages.cs:17]: Acknowledge of what?

Comment (MolaeMaceria) [reply source/StepBro.ExecutionHelper/ExecutionHelperMessages.cs:17]: Yup

Comment (MolaeMaceria) [reply source/StepBro.ExecutionHelper/ExecutionHelperMessages.cs:17]: Well, technically acknowledge that everything went well

**Review by JanJorgensen — APPROVED** [review]

**Pull request merged**